packages/deployer/src/compiler.ts

```diff
@@ -116,67 +150,51 @@ export class Compiler {
     public async compileAllAsync(): Promise<void> {
         await this._createArtifactsDirIfDoesNotExistAsync();
         this._contractSourcesIfExists = await Compiler._getContractSourcesAsync(this._contractsDir);
```

Comment:
If this can not exist, we should check for it's existence before using it. `IfExists` semantically means that this variable could be `undefined`.

Reply:
It doesn't matter in this case because it will be overwritten, though.

Reply:
Ah no, `_getContractSourcesAsync` returns an empty object, never `undefined`. So the variable here should simply be `this._contractSources` which may be an empty object.

Reply:
@abandeali1